extension/src/state_aggregate/rollup.rs

```diff
@@ -80,14 +80,18 @@ impl OwnedCompactStateAgg {
             "can't merge aggs with different state types"
         );
 
-        let (earlier, later) = match self.first_time.cmp(&other.first_time) {
+        let (earlier, later) = match self.cmp(&other) {
             Ordering::Less => (self, other),
             Ordering::Greater => (other, self),
-            Ordering::Equal => panic!("can't merge overlapping aggregates (same start time)"),
+            Ordering::Equal => panic!(
+                "can't merge overlapping aggregates (same start time: {})",
+                self.first_time
+            ),
         };
+
         assert!(
             earlier.last_time <= later.first_time,
-            "can't merge overlapping aggregates"
+            "can't merge overlapping aggregates (earlier={earlier:?}, later={later:?})"
```

Comment:
Might be useful to include both start and end times for the overlapping aggregates